lntest/harness.go

```diff
@@ -0,0 +1,828 @@
+package lntest
+
+import (
+	"fmt"
+	"io/ioutil"
+	"sync"
+	"time"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/grpclog"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/roasbeef/btcd/chaincfg"
+	"github.com/roasbeef/btcd/chaincfg/chainhash"
+	"github.com/roasbeef/btcd/integration/rpctest"
+	"github.com/roasbeef/btcd/rpcclient"
+	"github.com/roasbeef/btcd/txscript"
+	"github.com/roasbeef/btcd/wire"
+	"github.com/roasbeef/btcutil"
+)
+
+// NetworkHarness is an integration testing harness for the lightning network.
+// The harness by default is created with two active nodes on the network:
+// Alice and Bob.
+type NetworkHarness struct {
+	rpcConfig rpcclient.ConnConfig
+	netParams *chaincfg.Params
+	Miner     *rpctest.Harness
+
+	activeNodes map[int]*HarnessNode
+
+	// Alice and Bob are the initial seeder nodes that are automatically
+	// created to be the initial participants of the test network.
+	Alice *HarnessNode
+	Bob   *HarnessNode
+
+	bitcoinWatchRequests chan *txWatchRequest
+
+	// Channel for transmitting stderr output from failed lightning node
+	// to main process.
+	lndErrorChan chan error
+
+	quit chan struct{}
+
+	mtx sync.Mutex
+}
+
+// NewNetworkHarness creates a new network test harness. This initializes alice
+// and bob nodes given an already running instance of btcd's rpctest harness and
+// extra command line flags, which should be formatted properly - "--arg=value".
```

Comment:
Comment should be updated, this no longer takes the extra command line flags directly.